engine/src/main/java/org/pentaho/di/job/entries/simpleeval/JobEntrySimpleEval.java

```diff
@@ -0,0 +1,1172 @@
+              }
+              success = ( datevalue.equals( datecompare ) );
+            }
+            break;
+          case SUCCESS_NUMBER_CONDITION_NOT_IN_LIST: // not in list
+            if ( isDebug() ) {
+              logDebug( BaseMessages.getString(
+                PKG, "JobSimpleEval.Log.CompareWithValue", sourcevalue, realCompareValue ) );
+            }
+            realCompareValue = Const.NVL( realCompareValue, "" );
+            parts = realCompareValue.split( "," );
+            success = true;
+            for ( int i = 0; i < parts.length && success; i++ ) {
+              try {
+                datecompare = convertToDate( realCompareValue, realMask, df );
+              } catch ( Exception e ) {
+                logError( toString(), e.getMessage() );
+                return result;
+              }
+              success = ( !datevalue.equals( datecompare ) );
+            }
+            break;
+          default:
+            break;
+        }
+        df = null;
+        break;
+      case FIELD_TYPE_BOOLEAN:
+        boolean valuebool;
+        try {
+          valuebool = ValueMetaString.convertStringToBoolean( sourcevalue );
+        } catch ( Exception e ) {
+          logError( BaseMessages.getString( PKG, "JobEntrySimpleEval.Error.UnparsableBoolean", sourcevalue, e
+            .getMessage() ) );
+          return result;
+        }
+
+        switch ( successbooleancondition ) {
+          case SUCCESS_BOOLEAN_CONDITION_FALSE: // false
+            success = ( !valuebool );
+            break;
+          case SUCCESS_BOOLEAN_CONDITION_TRUE: // true
+            success = ( valuebool );
+            break;
+          default:
+            break;
+        }
+        break;
+      default:
+        break;
+    }
+
+    result.setResult( success );
+    // PDI-6943: this job entry does not set errors upon evaluation, independently of the outcome of the check
+    result.setNrErrors( 0 );
+    return result;
+  }
+
+  /*
+   * Returns variable with specifications
+   */
+  private String getVariableWithSpec() {
+    String variable = getVariableName();
+    if ( ( !variable.contains( StringUtil.UNIX_OPEN ) && !variable.contains( StringUtil.WINDOWS_OPEN ) && !variable
+      .contains( StringUtil.HEX_OPEN ) )
+      && ( ( !variable.contains( StringUtil.UNIX_CLOSE ) && !variable.contains( StringUtil.WINDOWS_CLOSE ) && !variable
+        .contains( StringUtil.HEX_CLOSE ) ) ) ) {
+      // Add specifications to variable
+      variable = StringUtil.UNIX_OPEN + variable + StringUtil.UNIX_CLOSE;
+      if ( isDetailed() ) {
+        logDetailed( BaseMessages.getString( PKG, "JobEntrySimpleEval.CheckingVariable", variable ) );
+      }
+    }
+    return variable;
+  }
+
+  private Date convertToDate( String valueString, String mask, SimpleDateFormat df ) throws KettleException {
+    Date datevalue = null;
+    try {
+      datevalue = df.parse( valueString );
+    } catch ( Exception e ) {

```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)